plugins/modules/foreman_search_facts.py

```diff
@@ -88,6 +93,17 @@
   register: result
 - debug:
     var: result.resources
+
+- name: Get all existing subscriptions
+  foreman_search_facts:
+    username: "admin"
+    password: "changeme"
+    server_url: "https://foreman.example.com"
+    resource: subscriptions
+    params: "organization_id=1"
```

Comment:
```suggestion
    params:
      organization_id: 1
```

Reply:
IMHO, it's true that users typically don't know these low-level details, such as org_id or scope of a particular call (nor should they). However, I feel this might be useful in addition to adding other parameters. I.e. it would be nice to add organization, but I suspect (and correct me if I'm wrong here) that there will be other, call-specific params. 

For example, I see `scope = {'template_id': job_template['id']}` in `foreman_job_template`. I can't test the module at this particular moment, but I suspect that if we wanted to list the job templates, we'd have to pass a template ID..?

So, I'm all for specific parameters (like `organization`), but isn't it beneficial to have a direct way to pass params to modules? (for other modules)

Reply:
Oh yeah, that's what I meant by "Or probably both, as organization is something that will happen often, but maybe the user will need to search for something else that requires more input than just the org?"!

Let's add `organization` (and probably `location`) explicitly, but still allow *anything* via `params`.

@sean797 @mdellweg @bagasse what do you think?

Reply:
@evgeni suggestion sounds sane to me! :+1:

Reply:
AS @evgeni stated, I think that we should not ask to users to know IDs of resources to use foreman ansible modules if there is a way to use name/title. But for this particular maybe managing mappings `X[s] -> X_id[s] -> find_resource_by_[name|title](Xs, ...)` for every possible relations will be a pain.

But as Katello resources are all scoped to an `organization`, :+1: to add it directly. For `location` OTOMH, I think there is no resources in foreman/katello that rely only on one location (only `locations` are used, but I can be wrong) so I don't think we should add a shortcut for `location`.

Reply:
While nothing *requires* location (as opposed to organization in Katello context), most entities can be scoped/assigned to a location, so if we look at `foreman_search_facts` as a generic "give me a list of entities, having the default scopes (org, loc) available sounds beneficial.

But as it's not *required*, I think we can add that in a different PR if anyone wants to.